File: Day 18/547-number-of-provinces.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:

    void dfs(vector<vector<int>> &isConnected, vector<bool> &visited, int node){
        visited[node] = true;

        for(int i=0; i<isConnected.size(); i++){
            if(isConnected[node][i] == 1 && !visited[i]) dfs(isConnected, visited, i);
        }
    }

    int findCircleNum(vector<vector<int>>& isConnected) {
        int count = 0;

        vector<bool> visited(isConnected.size(), false);

        for(int i=0; i<isConnected.size(); i++){
            if(!visited[i]){
                dfs(isConnected, visited, i);
                count++;
            }
        }

        return count;

    }
};

int main(){


    return 0;
}

File: Day 18/547-number-of-provinces.py
from typing import List

class Solution:
    def dfs(self, isConnected: List[List[int]], visited: List[bool], node: int) -> None:
        visited[node] = True

        for i in range(len(isConnected)):
            if isConnected[node][i] == 1 and not visited[i]:
                self.dfs(isConnected, visited, i)

    def findCircleNum(self, isConnected: List[List[int]]) -> int:
        count = 0
        visited = [False] * len(isConnected)

        for i in range(len(isConnected)):
            if not visited[i]:
                self.dfs(isConnected, visited, i)
                count += 1

        return count